### PR TITLE
test: Edge cases - Smart Contracts Service Tests Part 5

### LINKED ISSUE #19760: Edge cases - Smart Contracts Service Tests

### PR DESCRIPTION
**Pull request by vtronkov**
**Description**:
* Add various batch smart contract tests 

**Related issue(s)**:

Fixes #19760

### CODE REVIEW

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/opcodes/AtomicOpCodesSuite.java [hunk 1]
```diff
@@ -0,0 +1,523 @@
...
+                Map.of("atomicBatch.isEnabled", "true", "atomicBatch.maxNumberOfTransactions", "50"));
+        testLifecycle.doAdhoc(cryptoCreate(BATCH_OPERATOR).balance(ONE_MILLION_HBARS));
+        testLifecycle.overrideInClass(Map.of("contracts.evm.version", "v0.50"));
+        testLifecycle.doAdhoc(
+                // Multiple tests use the same contract, so we upload it once here
+                uploadInitCode(SELF_DESTRUCT_CALLABLE_CONTRACT));
+    }
+
+    @HapiTest
+    final Stream<DynamicTest> payableCreate2WorksAsExpected() {
```
Comment (lukelee-sl) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/opcodes/AtomicOpCodesSuite.java:102]: Can there some comments as to the purpose of these tests?  They seem to be testing various overwritten ops codes in the context of atomic transactions.  Why these op codes?  Are they the ones where Hedera has custom implementations?  Are we sure that the other op codes behave correctly?  
It is also helpful to avoid jargon such as "stackedFactory" unless there is a comment to explain.

Comment (vtronkov) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/opcodes/AtomicOpCodesSuite.java:102]: There is no particular reason to choose exactly these op codes. I was going to wrap all smart contract HAPI tests into an atomic batch, but we decided that this is too much. The idea of these tests is to have a broad range of atomic wrapped existing tests to verify that it will behave correctly.
